pkg/operator/ceph/controller/spec.go

```diff
@@ -75,22 +75,26 @@ var (
 	cronLogRotate = `
 CEPH_CLIENT_ID=%s
 PERIODICITY=%s
-LOG_ROTATE_CEPH_FILE=/etc/logrotate.d/ceph
-
-if [ -z "$PERIODICITY" ]; then
-	PERIODICITY=24h
-fi
+LOG_ROTATE_CEPH_FOLDER=/etc/logrotate.d/ceph
+DAEMON_LOG_FILE=/var/log/ceph/$CEPH_CLIENT_ID.log
+LOG_MAX_SIZE=%s
 
 # edit the logrotate file to only rotate a specific daemon log
 # otherwise we will logrotate log files without reloading certain daemons
 # this might happen when multiple daemons run on the same machine
-sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FILE"
+sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FOLDER"
+
+# replace default daily with give user input
+sed --in-place "s/daily/$PERIODICITY/g" "$LOG_ROTATE_CEPH_FOLDER"
+
+# append maxsize at 4th line ater the $PERIODICITY with 4spaces to maintain indentation
+sed --in-place "4i \ \ \ \ maxsize $LOG_MAX_SIZE" "$LOG_ROTATE_CEPH_FOLDER"
 
 while true; do
-	sleep "$PERIODICITY"
-	echo "starting log rotation"
-	logrotate --verbose --force "$LOG_ROTATE_CEPH_FILE"
-	echo "I am going to sleep now, see you in $PERIODICITY"
+	echo "Size of file $DAEMON_LOG_FILE to rotate: $(ls -lh --human-readable $DAEMON_LOG_FILE | awk '{print $5}')"
```

Comment:
Why not use `stat` instead of `ls`? Also `-h` is the same as `--human-readable`.

Reply:
done